mmdet/models/anchor_heads/yolact_head.py

```diff
@@ -0,0 +1,390 @@
+            mlvl_bboxes.append(bboxes)
+            mlvl_scores.append(scores)
+            mlvl_coeffs.append(coeff_pred)
+        mlvl_bboxes = torch.cat(mlvl_bboxes)
+        if rescale:
+            mlvl_bboxes /= mlvl_bboxes.new_tensor(scale_factor)
+        mlvl_scores = torch.cat(mlvl_scores)
+        mlvl_coeffs = torch.cat(mlvl_coeffs)
+
+        det_bboxes, det_labels, det_coeffs = fast_nms(mlvl_bboxes, mlvl_scores, mlvl_coeffs,
+                                                        cfg.score_thr, cfg.nms, cfg.max_per_img)
+        return det_bboxes, det_labels, det_coeffs
+
+    def get_anchors(self, featmap_sizes, img_metas, device):
+        """Get anchors according to feature map sizes.
+
+        Args:
+            featmap_sizes (list[tuple]): Multi-level feature map sizes.
+            img_metas (list[dict]): Image meta info.
+            device (torch.device | str): device for returned tensors
+
+        Returns:
+            tuple: anchors of each image, valid flags of each image
+        """
+        num_imgs = len(img_metas)
+        num_levels = len(featmap_sizes)
+
+        # since feature map sizes of all images are the same, we only compute
+        # anchors for one time
+        multi_level_anchors = []
+        multi_level_valid_flag = []
+        for i in range(num_levels):
+            anchors = self.make_priors(featmap_sizes[i], 
+                                        self.anchor_base_sizes[i], 
+                                        device=device)
+            multi_level_anchors.append(anchors)
+            multi_level_valid_flag.append(torch.ones(anchors.size(0), dtype=torch.bool))
+
+        anchor_list = [multi_level_anchors for _ in range(num_imgs)]
+        valid_flag_list = [multi_level_valid_flag for _ in range(num_imgs)]
+
+        return anchor_list, valid_flag_list
+
+    def make_priors(self, featmap_size, anchor_base_size, device):
+        """ Note that priors are [x,y,width,height] where (x,y) is the center of the box. """
+        prior_data = []
+
+        feat_h, feat_w = featmap_size
+
+        # Iteration order is important (it has to sync up with the convout)
+        for j, i in product(range(feat_h), range(feat_w)):
+            # +0.5 because priors are in center-size notation
+            x = (i + 0.5) / feat_w * self.img_size[1]
+            y = (j + 0.5) / feat_h * self.img_size[0]
+            
+            for ar in self.anchor_ratios:
+                for scale in self.anchor_scales:
+                    ar = sqrt(ar)
+
+                    w = scale * ar * anchor_base_size
+                    h = scale / ar * anchor_base_size
+                    
+                    # This is for backward compatability with a bug where I made everything square by accident
+                    if self.square_anchor:
+                        h = w
+
+                    # prior_data += [x, y, w, h]
+                    prior_data += [x-0.5*w, y-0.5*h, x+0.5*w, y+0.5*h]
```

Comment:
The comment seems wrong, should be [x1, y1, x2, y2]?
The coordinates in `make_priors` may not be consistent with other parts like [bbox2delta](https://github.com/open-mmlab/mmdetection/blob/4a4ffd4c8fac40b115bf4485b2fdaac29d89aa75/mmdet/core/bbox/transforms.py#L6) which is called at  https://github.com/open-mmlab/mmdetection/blob/4a4ffd4c8fac40b115bf4485b2fdaac29d89aa75/mmdet/core/anchor/anchor_target.py#L134
```
Code of bbox2delta:
px = (proposals[..., 0] + proposals[..., 2]) * 0.5
py = (proposals[..., 1] + proposals[..., 3]) * 0.5
pw = proposals[..., 2] - proposals[..., 0] + 1.0
ph = proposals[..., 3] - proposals[..., 1] + 1.0
```
In `bbox2delta`, given a anchor, the width should be x2 - x1 + 1.
If using the above coordinate, the width should be x2 - x1.